File: src/main/java/com/example/app/repository/UserRepository.java
package com.example.app.repository;

import com.example.app.utils.model.entities.User;
import jakarta.transaction.Transactional;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

@Transactional
@Repository
public interface UserRepository extends CrudRepository<User, Long> {

    boolean existsUserByLogin(String login);
    boolean existsUserByLoginOrEmailAndActivated(String login,String email, boolean activated);

    boolean existsUserByLoginAndPasswordAndEmailAndActivated(String login, String password,String email, boolean activated);

    User getUserByEmail(String email);

    User getUserByLogin(String login);

    boolean existsUserByActivationCode(String activationCode);

    User getUserByActivationCode(String code);

    @Modifying
    @Query("update User u set u.activated = :activated where u.activationCode = :activationCode")
    void updateActivatedByActivationCode(
            @Param("activated") boolean activated,
            @Param("activationCode") String activationCode
    );

}

File: src/main/kotlin/ru/kaplaan/repository/UserRepository.kt
package ru.kaplaan.repository

import jakarta.transaction.Transactional
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import ru.kaplaan.domain.entity.user.User

@Transactional
@Repository
interface UserRepository : JpaRepository<User, Long> {

    fun existsUserByUsername(username: String): Boolean

    fun existsUserByActivationCode(activationCode: String): Boolean

    fun existsUserByUsernameOrEmailAndActivated(username: String, email: String, activated: Boolean): Boolean

    fun existsUserByUsernameAndPasswordAndEmailAndActivated(
        username: String,
        password: String,
        email: String,
        activated: Boolean
    ): Boolean

    fun findByEmail(email: String): User?

    fun findByUsername(username: String): User?

    fun getUserByActivationCode(code: String): User?

    @Modifying
    @Query("update User u set u.activated = :activated where u.activationCode = :activationCode")
    fun updateActivatedByActivationCode(
        @Param("activated") activated: Boolean,
        @Param("activationCode") activationCode: String
    )

}
